### PR TITLE
feat: low-memory Forward XM (best-of-K) for pitch/variance Rectified Flow

### PR DESCRIPTION
**Pull request by KakaruHayate**
## Summary

Lands **low-memory Forward Explorative Modeling (XM, best-of-K)** for the Rectified Flow (reflow) pitch / variance predictors and full-depth acoustic reflow, rebased onto the current `integration-dual-timestep-stack` default branch.

This is the successor to the reverted #58/#59 (repeat-bin Forward XM). Instead of folding bins, it explores candidate flow noises and backprops only the closest candidate, matching the design of [Explorative Modeling (XM)](https://arxiv.org/abs/2607.27372).

## Key properties (by design)

- **No data collapse**: `run_reflow_xm` explores over the *noise latent only* — every batch element always participates in the final backward, so the effective backward data amount per step equals the K=1 baseline.
- **No k-fold VRAM**: candidates are evaluated in `no_grad` chunks (`xm_chunk_size`), then the winning noise is **replayed once with gradients** so upstream condition encoders stay trainable.
- **Shared timestep per original sample** across all candidates (determinism), and replay uses the exact same `t`/`noise` so the trained objective matches the explored one.
- Guard rails: raises if the backbone has dropout (`dropout_rate=0` required for exact replay) and validates config combinations.

## Config knobs

- Acoustic: `xm_best_of_k`, `xm_chunk_size` (full-depth reflow only; requires `use_shallow_diffusion=false`).
- Variance: `xm_pitch_best_of_k`, `xm_variance_best_of_k`, `xm_chunk_size` — independently enable exploration for pitch and variance predictors. K=1 preserves the standard objective.

## Files

- `modules/core/xm.py` — low-memory Forward XM runner (new)
- `modules/core/reflow.py` — `prepare_training_inputs` / `training_forward` / `p_losses(noise=)` (kept dual-timestep training path intact)
- `modules/losses/reflow_loss.py` — `reduction='none'` per-sample masked loss
- `modules/toplevel.py` — optional `pitch_predictor_fn` / `variance_predictor_fn` injection points
- `training/acoustic_task.py`, `training/variance_task.py` — XM wiring
- configs + templates — new hparams
- `tests/test_reflow_xm.py` — unit tests (replay with grads, padding, chunked XM, list-target multi-variance, mean semantics)

## Tests

`python -m unittest tests/test_reflow_xm.py` — 5 tests pass.

## Summary by CodeRabbit

* **New Features**
  * Added optional Explorative Modeling for Rectified Flow training.
  * Configure best-of-K candidate exploration and chunk size for acoustic, pitch, and variance prediction.
  * Supports separate exploration settings for pitch and variance predictors.
  * Added low-memory processing for candidate evaluation.

* **Configuration**
  * Default settings preserve existing single-candidate training behavior.
  * Added validation and guidance for compatible training options, including shallow diffusion requirements.

### CODE REVIEW

**Comment by KakaruHayate** [comment]
@coderabbitai review

**Comment by KakaruHayate** [comment]
@coderabbitai review

**Comment by KakaruHayate** [comment]
@coderabbitai review
